### PR TITLE
#24

### PR DESCRIPTION
**Pull request by nnascim**

### CODE REVIEW

File: docs/topics_subscribers.md [hunk 1]
```diff
@@ -0,0 +1,332 @@
...
+5. Configure your project connection.
+Add the folder name to the `less.config` file as an environment variable. The value of this variable should match the name of the application you want to connect to.
+```yaml
+env_vars:
+  - my-less-project
+```
+
+Export the name of the application (deployment) you want to connect to. This allows you to point your application to different deployments (development, stage, production, etc.).
+```bash
+export my-less-project=my-less-project
```
Comment (RicardoJRR-23) [inline docs/topics_subscribers.md:321]: This is not a valid name format for a env var due the use of hyphen. And also the name for all the env var now should be in upper case, so the name of the env var on uppercase will match the name of the project

File: docs/topics_subscribers.md [hunk 2]
```diff
@@ -0,0 +1,332 @@
...
+  exports.process = async (request, response) => {
+    // Get the user payload from the body.
+    const user = JSON.parse(request.body);
+    
+    // Pretend to create a new user.
+    const new_user = { id: 'my-fake-id', ...user };
+
+    // Publish the new user to the `user_created` topic.
+    await topics.user_created.publish(
+      JSON.stringify(new_user)
```
Comment (RicardoJRR-23) [inline docs/topics_subscribers.md:106]: You don't need to make stringify of a message before publishing

File: docs/cron-jobs.md [hunk 3]
```diff
@@ -10,7 +10,7 @@ import TabItem from '@theme/TabItem';
 
 ## Create CRON Job
```
Comment (savylopes) [inline docs/cron-jobs.md:12]: @nnascim Here we also need to allow the cron expression variable in the less.config file. maybe we should mention it in this cron section

**Review by RicardoJRR-23 — CHANGES_REQUESTED** [review]

File: docs/quick-start.md [hunk 4]
```diff
@@ -9,17 +9,18 @@ import TabItem from '@theme/TabItem';
 
 **TLDR; Create a `GET /hello` route and deploy.**
```
Comment (savylopes) [inline docs/quick-start.md:10]: @nnascim is this "TLDR" a mistake?

File: docs/cron-jobs.md (hunk 3 above)
Comment (nnascim) [reply docs/cron-jobs.md:12]: @savylopes it's solved on line 37.

File: docs/rest-apis.md
Comment (savylopes) [inline docs/rest-apis.md:1]: @nnascim maybe we can show examples of how to access the created routes with curl. To facilitate the users

File: docs/rest-apis.md [hunk 5]
```diff
@@ -45,115 +49,218 @@ Let's create a `GET` request for the `/hello` route.
...
+    response.statusCode = 200;
+    return response;
+  };
   ```
   </TabItem>
   
 </Tabs>
 
-## HTTP Request
+
```
Comment (savylopes) [inline docs/rest-apis.md:169]: @nnascim we can add example of how to deal with multiple values for a query paramter

File: docs/shared-modules.md
Comment (savylopes) [inline docs/shared-modules.md:1]: @nnascim few comments  here:
1. I believe that we should mention that if we don't create the index.js file the folder will be ignored and node used as a package
2. we have to way to import things on shared. By create a empty index.js file on the folder that in this case will allow use to import the User model by doing `const User = require('models/User');` or by create a index.js file and explicitly import the User model and export it later on the file. On this way the user will be allowed to inmport it like on the example by doing `const { User } = require('models');`  

> 

>

File: docs/sign-in-sign-up.md
Comment (savylopes) [inline docs/sign-in-sign-up.md:1]: this step "[less-cli] Account verified! Please check your email for your Less credentials." do not happen now

File: docs/rest-apis.md
Comment (savylopes) [inline docs/rest-apis.md:1]: It is missing middleware documentation

File: docs/topics_subscribers.md [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+2. Using the Less Topics REST API.
+
+### Send Messages Using the SDK
+Use the Less `topics` module to send messages to all of a topic's subscribers.
+
+<Tabs groupId="programming-language" queryString="programming-language">
+  <TabItem value="nodejs" label="Node.js">
+    ```js showLineNumbers
+    const { topics } = require('@chuva.io/less');
+
```
Comment (savylopes) [inline docs/topics_subscribers.md:78]: I think you can create user object here like ex: `const user = {name: "John", email: "John@exampe.com"};` use it on the publish method bellow to be more clear an consistent with what the processor created above are expecting

File: docs/rest-apis.md
Comment (FredSoares) [inline docs/rest-apis.md:1]: @nnascim I think you should document how to use middleware

Comment (nnascim) [reply docs/rest-apis.md:1]: I've created a new issue for this

Comment (nnascim) [reply docs/rest-apis.md:1]: I've created a separate [issue](https://github.com/chuva-io/less-site/issues/26) for it.

File: docs/rest-apis.md (hunk 5 above)
Comment (nnascim) [reply docs/rest-apis.md:169]: I created [this ticket](https://github.com/chuva-io/less-site/issues/27) for it.

File: docs/shared-modules.md
Comment (nnascim) [reply docs/shared-modules.md:1]: I created [this ticket](https://github.com/chuva-io/less-site/issues/28) for it.

File: docs/topics_subscribers.md (hunk 6 above)
Comment (nnascim) [reply docs/topics_subscribers.md:78]: I don't want to add it here because I just want to show the function and import. There is example code below.

File: docs/topics_subscribers.md [hunk 7]
```diff
@@ -0,0 +1,330 @@
...
+```
+
+2. Add the projects you want to connect to to the `external_topics` folder.
+```bash
+mkdir less/external_topics/my-less-project
+```
+
+3. Add the topic want to subscribe to to the respective project folder.
+```bash
+mkdir less/external_topics/my-less-project/user_created
```
Comment (RicardoJRR-23) [inline docs/topics_subscribers.md:292]: We will not allow the use of hyphen on any resource

File: docs/deploy.md [hunk 8]
```diff
@@ -41,53 +41,59 @@ Deploying the following structure would give you the output below.
...
+[less-cli] Deploying... 🚀
+[less-cli] Deployment complete ✅
+[less-cli] Resources
+[less-cli]   - API URLs
+[less-cli]     - chat: https://[PROJECT_NAME]-chat.api.eu-0.a83b464c9.less.chuva.cv
+[less-cli]     - webhooks: https://[PROJECT_NAME]-webhooks.api.eu-0.a83b464c9.less.chuva.cv
+[less-cli]     - topics: https://[PROJECT_NAME]-topics.api.eu-0.a83b464c9.less.chuva.cv
+[less-cli]     - functions: https://[PROJECT_NAME]-functions.api.eu-0.a83b464c9.less.chuva.cv
+[less-cli]   - Web Socket URLs
+[less-cli]     - realtime_chat: wss://[PROJECT_NAME]-realtime-chat.api.eu-0.a83b464c9.less.chuva.cv
```
Comment (RicardoJRR-23) [inline docs/deploy.md:67]: on socket  realtime_chat instead of `.api.` on the url it should be `.ws.`

Comment (nnascim) [reply docs/deploy.md]: Thanks. Fixed.

File: docs/topics_subscribers.md (hunk 7 above)
Comment (nnascim) [reply docs/topics_subscribers.md]: Thanks. Fixed.

**Review by RicardoJRR-23 — APPROVED** [review]

**Review by savylopes — APPROVED** [review]
